### DISCUSSION TITLE
Automatic trimming?

### DISCUSSION THREAD

**Discussion by tiloc** [renggli/dart-xml#120] [Q&A] [answered]
I am parsing XML with lots of text nodes. It is split out into multiple lines. Like
<node>
    Text goes here...
</node>

Now when I access the .text property of a node to get the text it comes with a lot of leading and trailing line breaks and tabs. Of course I can call trim() on each and every one, but could the parser get an option to automatically return trimmed text?

**Comment by renggli** [comment, marked as answer]
You can easily do that as part of a transformation step after parsing:

```dart
final document = XmlDocument.parse(input);

// Trim all the text nodes
for (final node in document.descendants.whereType<XmlText>()) {
  node.replace(XmlText(node.text.trim()));
}
```

Possibly you also want to normalize the document afterwards (remove empty text nodes): 

```dart
document.normalize();
```

**Comment by tiloc** [comment]
Thanks! That works great for me!
